mne_bids/write.py

```
@@ -1352,3 +1353,162 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
+        discarding their descriptions.
+    verbose : bool
+        The verbosity level.
+
+    Raises
+    ------
+    ValueError
+        If the number of channels and descriptions does not match.
+
+    ValueError
+        If the specified channels cannot be found in the dataset.
+
+    ValueError
+        If the specified ``kind`` cannot be found in the dataset.
+
+    ValueError
+        If ``overwite=False``, but no channel names were passed.
+
+    RuntimeError
+        If multiple recording kinds are present in the dataset, but
+        ``kind=None``.
+
+    RuntimeError
+        If more than one data files exist for the specified recording.
+
+    Examples
+    --------
+    Mark a single channel as bad.
+
+    >>> mark_bad_channels('MEG 0112', bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark multiple channels as bad.
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> mark_bad_channels(bads, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark channels as bad, and add a description as to why.
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> descriptions = ['Only produced noise', 'Continuously flat']
+    >>> mark_bad_channels(bads, descriptions, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark channels as bad, and reset all others existing in the dataset to
+    "good".
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> mark_bad_channels(bads, overwrite=True, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark all channels as good.
+
+    >>> mark_bad_channels([], bids_basename=bids_basename, bids_root=bids_root)
+
+    """
+    if not channels and not overwrite:
+        raise ValueError('You did not pass a channel name, but set '
+                         'overwrite=False. If you wish to mark all channels '
+                         'as good, please pass overwrite=True')
```

maybe this will be annoying in a for loop but let's see if people complain

---

I'm open for suggestions to change this behavior; I just feel that it's unclear what the user wants in this case, so I thought it would be better to raise an exception instead of making assumptions

---

ok let's keep like this for now